### ISSUE TITLE
Feature Match loss Issue

### ISSUE THREAD

**Issue by wftang2020** [NVIDIA/pix2pixHD#237]
hi: @tcwang0509
     if not self.opt.no_ganFeat_loss:
            feat_weights = 4.0 / (self.opt.n_layers_D + 1)
            D_weights = 1.0 / self.opt.num_D
            for i in range(self.opt.num_D):
                for j in range(len(pred_fake[i])-1):
                    loss_G_GAN_Feat += D_weights * feat_weights * \
                        self.criterionFeat(pred_fake[i][j], pred_real[i][j].detach()) * self.opt.lambda_feat
     the paper mentioned that Dk only serves as a feature extractor and does not maximize the loss LFM in feature matching loss. but when I read the code, I see the last line only pred_real[i][j] using detach. Is it correct? Does "not maximize the loss" here mean the D not participate in autograd.

**Comment by Claire210222** [comment]
Have you solved the problem? I have the same question

**Comment by victorca25** [comment]
You need to detach the target image logits, you only backpropagate the fake's. The code is correct.
